Updated and created new tests for the integration of npc and npc_move modules

torresszy opened the pull request:
This pull request relies on pull request #904 to be merged successfully.

After integrating the `npc_mov` struct in `npc_move.c` into the `npc` struct in `npc.c`, I have updated the relevant tests to make sure that this new field in the `npc` struct is covered in the tests as well. I also refactored `test_rooms_npc.c` and moved all the tests related to `npc_mov` into the new `test_npc_move.c` file.

Summary of major changes:

Refactored `test_rooms_npc.c` moved `npc_mov` related tests into the new `test_npc_move.c` file.
Updated `test_npc.c` so the new field, `npc_mov_t` will be tested now as well.
Added a new test for the function, `delete_npc_from_room` that was implemented in pull request #904. This test is added in the file `test_rooms_npc.c`
Updated the CMakefile to include the new `test_npc_move.c`

mbutera17 reviewed (approved):
Everything looks good! I had read through most of this when the previous pull request was made (specifically when we divided the module). The tests look good and changes to any other files (quests files, example files, etc.) make sense to me.

torresszy commented:
The pull request #904 has been merged successfully, and the merge conflicts resulting from that pull request have been resolved. This pull request is now ready to be reviewed and merged

lilyehsani commented on tests/npc/test_rooms_npc.c:
```diff
@@ -135,6 +82,39 @@ Test (npcs_in_room, add_npc_to_room)
+/* Tests delete_npc_from_room function */
+Test (npcs_in_room, delet_npc_from_room)
+{
+    //stats_t *stats = stats_new("npc_test_stat", 20);
```
Delete commented code or explain why it's commented.

torresszy replied:
This commented code has been deleted

lilyehsani reviewed (changes requested):
Looks really good! Added one comment.

lilyehsani reviewed (approved):
LGTM! Approved.

torresszy commented:
This is the closing comment for this pull request; it has been merged successfully and all the goals achieved. No major changes were needed since the beginning of this pull request. There was a minor merge conflict due to changes in `npc.h` from another pull request, but it was resolved very quickly. 

One note for future implementation is that I have refactored `test_rooms_npc.c` and moved `npc_mov` related tests into the new` test_npc_move.c` file. However, since the `rooms-npc` module is likely to be integrated in to `room.h` module completely, `test_rooms_npc.c` should also be updated and refactored into`test_rooms.c` eventually.

lilyehsani commented:
Issue Score: ✔️++

Comments:
Great work overall! The summary and closing of this PR are very thorough.

<!-- {"score": "check-plus-plus"} -->
